Automattic/newspack-content-converter
# PHPCS and JSPrettier

This PR adds syntax and code standard enforcement to the plugin to match other Newspack plugins. The additions are PHPCS and JSPrettier. After installing and configuring these, I took the following steps:

- Ran all .js and .css files through JSPrettier. All of the changes to these files should be syntax only, with no functional changes.
- For some reason, JSPrettier was not behaving as expected with `src/lib/converter.js`. Although I'm not positive, I suspect this was some limitation on scripts that reside in a `lib` directory, so I renamed the file to `src/converter/index.js` and the problem went away. 
- Went through all the .php files and made all non-controversial PHPCS changes (syntax, docblocks, etc).
- There are a couple of deeper PHPCS warnings that may require a bit more analysis, so I left these alone for now. These included the instantiation of `NewspackContentConverter`, file operations not using `WP_Filesystem`, missing nonce verification, null coalescing operator, and `$_GET` super global. 

### Testing Instructions

1. Run `npm i` and `composer install`
2. Verify the plugin works exactly as before — nothing in this PR should change any functionality.
